ext/standard/password.c

```diff
@@ -487,46 +487,45 @@ const php_password_algo* php_password_algo_find(const zend_string *ident) {
 	return Z_PTR_P(tmp);
 }
 
-static const php_password_algo* php_password_algo_find_zval_ex(zval *arg, const php_password_algo* default_algo) {
-	if (!arg || (Z_TYPE_P(arg) == IS_NULL)) {
+static const php_password_algo* php_password_algo_find_zval_ex(
+	zend_string *arg_str, zend_long arg_long, zend_bool arg_is_null, const php_password_algo* default_algo
+) {
+	if (arg_is_null) {
 		return default_algo;
 	}
 
-	if (Z_TYPE_P(arg) == IS_LONG) {
-		switch (Z_LVAL_P(arg)) {
-			case 0: return default_algo;
-			case 1: return &php_password_algo_bcrypt;
+	if (arg_str) {
+		return php_password_algo_find(arg_str);
+	}
+
+	switch (arg_long) {
+		case 0: return default_algo;
+		case 1: return &php_password_algo_bcrypt;
 #if HAVE_ARGON2LIB
-			case 2: return &php_password_algo_argon2i;
-			case 3: return &php_password_algo_argon2id;
+		case 2: return &php_password_algo_argon2i;
+		case 3: return &php_password_algo_argon2id;
 #else
-			case 2:
-				{
-				zend_string *n = zend_string_init("argon2i", sizeof("argon2i")-1, 0);
-				const php_password_algo* ret = php_password_algo_find(n);
-				zend_string_release(n);
-				return ret;
-				}
-			case 3:
-				{
-				zend_string *n = zend_string_init("argon2id", sizeof("argon2id")-1, 0);
-				const php_password_algo* ret = php_password_algo_find(n);
-				zend_string_release(n);
-				return ret;
-				}
+		case 2:
+			{
+			zend_string *n = zend_string_init("argon2i", sizeof("argon2i")-1, 0);
+			const php_password_algo* ret = php_password_algo_find(n);
+			zend_string_release(n);
+			return ret;
+			}
+		case 3:
+			{
+			zend_string *n = zend_string_init("argon2id", sizeof("argon2id")-1, 0);
+			const php_password_algo* ret = php_password_algo_find(n);
+			zend_string_release(n);
+			return ret;
+			}
 #endif
-		}
-		return NULL;
-	}
-
-	if (Z_TYPE_P(arg) != IS_STRING) {
-		return NULL;
 	}
 
-	return php_password_algo_find(Z_STR_P(arg));
+	return NULL;
 }
-static const php_password_algo* php_password_algo_find_zval(zval *arg) {
-	return php_password_algo_find_zval_ex(arg, php_password_algo_default());
+static const php_password_algo* php_password_algo_find_zval(zend_string *arg_str, zend_long arg_long, zend_bool arg_is_null) {
+	return php_password_algo_find_zval_ex(arg_str, arg_long, arg_is_null, php_password_algo_default());
```

Comment:
php_password_algo_find_zval_ex is used only here, I'd inline it.

Reply:
All done!